notebooks/official/custom/custom_training_tensorboard_profiler.ipynb

```diff
@@ -0,0 +1,1639 @@
+{
```

Comment:
<p>Your training code must be configured to write TensorBoard logs to a Cloud Storage bucket, the location of which Vertex AI Training automatically makes available through a predefined environment variable, <code>AIP_TENSORBOARD_LOG_DIR</code>.</p><p> </p><p>This can usually be done by providing <code>os.environ['AIP_TENSORBOARD_LOG_DIR']</code> as the log directory to the open source TensorBoard log writing APIs.</p><p> </p><p>For example, in TensorFlow 2.x, you can use following code to create a tensorboard_callback:</p><p> </p><pre class="ql-syntax" spellcheck="false">tensorboard_callback = tf.keras.callbacks.TensorBoard(  
  log_dir=os.environ['AIP_TENSORBOARD_LOG_DIR'],  
  histogram_freq=1)  
</pre><p> </p><p><code>AIP_TENSORBOARD_LOG_DIR</code> is in the <code>BASE_OUTPUT_DIR</code> that you provide when creating the custom training job.</p><p> </p><p>To enable Vertex AI TensorBoard Profiler for your training job, add the following to your training script:</p><p> </p><p>Add the <code>cloud_profiler</code> import at your top level imports:</p><p> </p><pre class="ql-syntax" spellcheck="false">from google.cloud.aiplatform.training_utils import cloud_profiler 
</pre><p> </p><p>Initialize the cloud_profiler plugin by adding:</p><p> </p><pre class="ql-syntax" spellcheck="false">cloud_profiler.init()
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/719/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='75'/>